docs/secure-communication-agents.asciidoc

```diff
@@ -0,0 +1,89 @@
+. <<https-in-agents,Enable HTTPS in your APM Agents>>
+
+[[secret-token]]
+[float]
+=== Secret token
+
+You can configure a secret token to authorize requests to the APM Server.
+This ensures that only your agents are able to send data to your APM servers.
+Both the agents and the APM servers have to be configured with the same secret token,
+and secret tokens only provide security when used in combination with SSL/TLS.
+
+**APM Server configuration**
+
+Here's how you set the secret token in APM Server:
+
+[source,yaml]
+----
+apm-server.secret_token: <secret-token>
+----
+
+We recommend saving the token in the APM Server <<keystore>>.
+
+IMPORTANT: Secret tokens are not applicable for the RUM Agent,
+as there is no way to prevent them from being publicly exposed.
+
+**Agent specific configuration**
+
+Each Agent has a configuration for setting the value of the secret token:
+
+* *Go Agent*: {apm-go-ref}/configuration.html#config-secret-token[`ELASTIC_APM_SECRET_TOKEN`]
+* *Java Agent*: {apm-java-ref}/config-reporter.html#config-secret-token[`secret_token`]
+* *.NET Agent*: {apm-dotnet-ref}/config-reporter.html#config-secret-token[`ELASTIC_APM_SECRET_TOKEN`]
+* *Node.js Agent*: {apm-node-ref}/configuration.html#secret-token[`Secret Token`]
+* *Python Agent*: {apm-py-ref}/configuration.html#config-secret-token[`secret_token`]
+* *Ruby Agent*: {apm-ruby-ref}/configuration.html#config-secret-token[`secret_token`]
+
+[[ssl-setup]]
+[float]
+=== SSL/TLS communication in APM Server
+
+To enable SSL/TLS, you need both a private key and a certificate issued by a certification authority (CA).
+You can then specify the path to those files in your configuration properties.
+This will make APM Server serve HTTPS requests instead of HTTP.
+
+Here's a basic output config with secure communication enabled:
+
+[source,yaml]
+----
+apm-server.ssl.key: "/etc/pki/key.pem"
+apm-server.ssl.certificate: "/etc/pki/client.pem"
+output.elasticsearch.ssl.certificate_authorities: ["/etc/pki/root/ca.pem"]
+----
+
+A full list of configuration options is available in <<agent-server-ssl>>.
+
+[[https-in-agents]]
+[float]
+=== HTTPS communication in APM Agents
+
+To enable secure communication in your Agents, you simply need to update the configured server URL to use `HTTPS` instead of `HTTP`.
+
+* *Go Agent*: {apm-go-ref}/configuration.html#config-server-url[`ELASTIC_APM_SERVER_URL`]
+* *Java Agent*: {apm-java-ref}/config-reporter.html#config-server-urls[`server_urls`]
+* *.NET Agent*: {apm-dotnet-ref}/config-reporter.html#config-server-urls[`ServerUrls`]
+* *Node.js Agent*: {apm-node-ref}/configuration.html#server-url[`serverUrl`]
+* *Python Agent*: {apm-py-ref}/[`server_url`]
+* *Ruby Agent*: {apm-ruby-ref}/configuration.html#config-server-url[`server_url`]
+
+Some Agents also allow you to specify a custom certificate authority for connecting to APM Server.
+This allows you to enable encryption and verify that you are sending data to the correct APM Server.
```

Comment:
> This allows you to enable encryption and verify that you are sending data to the correct APM Server.

I would remove this sentence. You can still verify the server when not configuring a custom CA but using a trusted one, but only for certs signed by a trusted CA. The custom CA config allows you to use and verify self signed certs.